kaspi_app/SwiftUI/ProductSwiftUI.swift

```diff
@@ -0,0 +1,184 @@
+                    
+                }
+            }
+        }
+    }
+}
+
+
+struct ProductView: View {
+    let product:Product
+    let type: ProductCardType
+    var heigt: CGFloat{
+        switch type {
+        case .large:
+            return 233
+        case .small:
+            return 195
+        }
+    }
+    var width: CGFloat{
+        switch type {
+        case .large:
+            return UIScreen.main.bounds.size.width/2-31
+        case .small:
+            return 108
+        }
+    }
+    
+    var body: some View {
+        Button(action: product.action){
+            VStack(alignment: .leading){
+                ZStack{
+                    product.image
+                        .resizable()
+                        .scaledToFit()
+                    VStack(spacing: 6){
+                        Text(product.bonus)
+                            .multilineTextAlignment(.center)
+                            .frame(width: 38, height: 17)
+                            .background(Rectangle().fill(Color(red: 0.35, green: 0.62, blue: 0.02)).cornerRadius(8))
+                            .font(Font.system(size: 12))
+                            .foregroundColor(.white)
+                        if product.credit0012{
+                            Text("0:0:12")
+                                .multilineTextAlignment(.center)
+                                .frame(width: 38, height: 17)
+                                .background(Rectangle().fill(.red).cornerRadius(8))
+                                .font(Font.system(size: 12))
+                                .foregroundColor(.white)
+                        }
+                    }.position(x:16,y: 20)
+                }
+            
+                Text(product.name)
+                    .font(Font.system(size: 13))
+                    .foregroundColor(.black)
+                    .lineLimit(1)
+                Text(product.category)
+                    .font(Font.system(size: 13))
+                    .foregroundColor(Color(red: 0.64, green: 0.64, blue: 0.64))
+                
+                switch type {
+                case .large:
+                    HStack{
+                        Text(getStringFromInt(int:product.price))
+                            .font(Font.system(size: 13,weight: .bold))
+                            .foregroundColor(.black)
+                           
+                        HStack(spacing:3){
+                            Text(getStringFromInt(int:product.price/6))
+                                .font(Font.system(size: 12,weight: .bold))
+                                .foregroundColor(.black)
+                                .background(Rectangle().fill(Color(red: 1, green: 0.81, blue: 0)))
+                            Text("x6")
+                                .font(Font.system(size: 13))
+                                .foregroundColor(Color(red: 0.64, green: 0.64, blue: 0.64))
+                        }
```

Comment:
Here you can move this code to separate variables in the bottom of struct, and use these variables instead of repeating the same code twice:

case .large:
     HStack{
         text // you can name as you want
         secondaryText
     }
case .small:
      text // you can name as you want
      secondaryText